aspnetcore/security/authentication/cookie.md

```diff
@@ -14,173 +14,299 @@ uid: security/authentication/cookie
 
-<a name=security-authentication-cookie-middleware-configuring></a>
+<a name="security-authentication-cookie-middleware-configuring"></a>
 
 ## Adding and configuring
 
-The first step is adding the cookie middleware to your application. First use nuget to add the `Microsoft.AspNetCore.Authentication.Cookies` package. Then add the following lines to the `Configure` method in your *Startup.cs* file before the `app.UseMvc()` statement;
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-```csharp
-app.UseCookieAuthentication(new CookieAuthenticationOptions()
-   {
-       AuthenticationScheme = "MyCookieMiddlewareInstance",
-       LoginPath = new PathString("/Account/Unauthorized/"),
-       AccessDeniedPath = new PathString("/Account/Forbidden/"),
-       AutomaticAuthenticate = true,
-       AutomaticChallenge = true
-   });
-   ```
+Complete the following steps:
+
+- Install the `Microsoft.AspNetCore.Authentication.Cookies` NuGet package in your project. This package contains the cookie middleware.
+
+- Add the following lines to the `Configure` method in your *Startup.cs* file before the `app.UseMvc()` statement:
+
+    ```csharp
+    app.UseCookieAuthentication(new CookieAuthenticationOptions()
+    {
+        AccessDeniedPath = "/Account/Forbidden/",
+        AuthenticationScheme = "MyCookieMiddlewareInstance",
+        AutomaticAuthenticate = true,
+        AutomaticChallenge = true,
+        LoginPath = "/Account/Unauthorized/"
+    });
+    ```
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+Complete the following steps:
 
-The code snippet above configures a few options;
+- If targeting .NET Framework, install version 2.0+ of the `Microsoft.AspNetCore.Authentication.Cookies` NuGet package in your project.
 
-* AuthenticationScheme - this is a value by which the middleware is known. This is useful when there are multiple instances of middleware and you want to [limit authorization to one instance](../authorization/limitingidentitybyscheme.md#security-authorization-limiting-by-scheme).
+- Invoke the `UseAuthentication` method in the `Configure` method of the *Startup.cs* file:
 
-* LoginPath - this is the relative path requests will be redirected to when a user attempts to access a resource but has not been authenticated.
+    ```csharp
+    app.UseAuthentication();
+    ```
 
-* AccessDeniedPath - this is the relative path requests will be redirected to when a user attempts to access a resource but does not pass any [authorization policies](../authorization/policies.md#security-authorization-policies-based) for that resource.
+- Invoke the `AddAuthentication` and `AddCookie` methods in the `ConfigureServices` method of the *Startup.cs* file:
 
-* AutomaticAuthenticate - this flag indicates that the middleware should run on every request and attempt to validate and reconstruct any serialized principal it created.
+    ```csharp
+    services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
```

Comment:
This should use "MyCookieMiddlewareInstance" to match the 1.x sample and the SignIn samples.